src/patterns/dosdp-patterns/response_to_chemical_with_role_stimulus.yaml

```diff
@@ -0,0 +1,62 @@
+---
+pattern_name: response_to_chemical_with_role_stimulus
+
+pattern_iri: http://purl.obolibrary.org/obo/oba/patterns/response_to_chemical_with_role_stimulus.yaml
+
+description: "A trait that affects the biological response to a representative
+  or representatives of a chemical entity group where the members of the group
+  share a role which they can exhibit. Examples of roles include 'drug',
+  'protease inhibitor', 'opioid agent', etc."
+
+contributors:
+  - https://orcid.org/0000-0001-8314-2140  # Ray Stefancsik
+  - https://orcid.org/0000-0002-7356-1779  # Nicolas Matentzoglu
+  - https://orcid.org/0000-0002-7073-9172  # David Osumi-Sutherland
+
+classes:
+  role: CHEBI:50906
+  process_quality: PATO:0001236
+  response_to_stimulus: GO:0050896
+
+
+relations:
+  characteristic_of: RO:0000052
+  has_input: RO:0002233
+  has_role: RO:0000087
+
+annotationProperties:
+  exact_synonym: oio:hasExactSynonym
+  contributor: dct:contributor  # contributor: dcterms:contributor
+
+vars:
+  role: "'role'"
+
+data_list_vars:
+  orcid: xsd:string  # ORCID of contributor
+
+name:
+  text: "trait in response to %s"
+  vars:
+    - role
+
+annotations:
+  - annotationProperty: exact_synonym
+    text: "response to %s trait"
+    vars:
+      - role
+
+  - annotationProperty: contributor
+    value: orcid
+
+def:
+  text: "A trait that affects the response to a stimulus with %s."
+  vars:
+    - role
+
+equivalentTo:
+  text: "'process_quality' and
+        ('characteristic_of' some
+        ('response_to_stimulus' and
+        ('has_input' some ('has_role' some %))))"
```

Comment:
Just to be fully aligned with upheno:

https://raw.githubusercontent.com/obophenotype/upheno/master/src/patterns/dosdp-dev/abnormalLevelOfChemicalEntityWithRoleInLocation.yaml

```
('has_input' some ('chemical entity' and ('has_role' some %s)))))
```
?